### PR TITLE
Staged Payments API: Releasing staged payments 

### LINKED ISSUE #2486: Staged Payments API: Releasing staged payments
**Releasing staged payments (as owner)**
- Saga should accept array of slot IDs and send a multicall of `releaseStagedPayment`
- Claiming step can be removed as it is done by contracts 

**Releasing staged payment via arbitration**
- `releaseExpenditureStage` saga should be modified to use `releaseStagedPaymentViaArbitration` if caller is not the owner

**Release staged payment via motion**
- Saga should accept array of slot IDs and create a motion to send a multicall of `releaseStagedPayment`
- Schema needs to be modified to store an array of slot IDs

- `isReleased` field can be removed from metadata. The preferred way of determining if stage is released is by checking all slot payouts `isClaimed` property

### PR DESCRIPTION
**Pull request by jakubcolony**
## Description

Please see issue for description.

Additionally, this PR fixes a regression introduced by myself some time ago where staged payments did not have their claim delays set to max possible value (and were therefore "released" straight after creation).

## Testing

Make sure you have Staged Expenditure and Voting Reputation extensions enabled.

**Setting correct claim delay on payment creation**
There was a regression that caused the stages to be immediately claimable.
- Create a staged payment
- Confirm the claim delay for all slots is set to max uint256 value

**Releasing staged payments (as owner)**
- Create a staged payment - you can adjust the parameters in `TmpAdvancedPayments`, by default there'll be 2 slots
- Lock, fund and finalize
- Test releasing single and multiple stages, confirm the funds have been paid out and run the query below to verify the data

**Releasing staged payment via arbitration**
- Create a staged payment as someone else
- Lock, fund and finalize
- As a user with `Arbitration` permissions, test releasing single and multiple stages

**Release staged payment via motion**
- Create a staged payment, lock, fund and finalize it
- Create a motion to release single and multiple stages, have it finalize and confirm the stages have been released

```gql
query ListStagedPayments {
  listExpenditures(filter: {type: {eq: STAGED}}) {
    items {
      nativeId
      slots {
        id
        payouts {
          amount
          isClaimed # should be `true` after releasing
        }
        claimDelay # max uint256 before releasing, 0 after
      }
      firstEditTransactionHash
      actions {
        items {
          id
          type
          expenditureSlotIds
          motionData {
            expenditureSlotIds
          }
        }
      }
    }
  }
}

```

There's a bunch of additional changes/refactoring in [block-ingestor](https://github.com/JoinColony/block-ingestor/pull/243).

TODO:
- [x] Clean up `ExpenditureStage` GraphQL type
- [x] Fix issue with claim delay not being set to max value for staged payments

Resolves #2486

### CODE REVIEW

File: src/redux/sagas/expenditures/releaseExpenditureStages.ts [hunk 1]
```diff
@@ -55,30 +59,74 @@ function* releaseExpenditureStage({
...
+      );
+
+    let userPermissionDomainId;
+    let userChildSkillIndex;
+    if (expenditure.ownerAddress !== userAddress) {
+      [userPermissionDomainId, userChildSkillIndex] = yield getPermissionProofs(
+        colonyClient.networkClient,
+        colonyClient,
+        expenditure.nativeDomainId,
+        ColonyRole.Administration,
```
Comment (iamsamgibbs) [inline src/redux/sagas/expenditures/releaseExpenditureStages.ts:83]: I'm guessing this is intended to be `Arbitration` as the function call is `releaseStagedPaymentViaArbitration`? I also tried testing as a user with only `Administration` to be sure and the contract call failed.

File: src/redux/sagas/utils/expendituresMulticall.ts [hunk 2]
```diff
@@ -22,6 +22,8 @@ const EXPENDITURESLOTS_SLOT = toB32(BigNumber.from(26));
 const EXPENDITURESLOT_RECIPIENT = toB32(BigNumber.from(0));
 const EXPENDITURESLOT_CLAIMDELAY = toB32(BigNumber.from(1));
 
+const MAX_CLAIM_DELAY_VALUE = BigNumber.from(2).pow(128).sub(1);
```
Comment (iamsamgibbs) [inline src/redux/sagas/utils/expendituresMulticall.ts:25]: Could use ethers `constants.MaxUint128` for this

**Review by iamsamgibbs — CHANGES_REQUESTED** [review]
Nice work on this Jakub! It is all working great, except for one tiny issue with permission roles - I've left a comment.

I was able to create a staged payment:

Screenshot 2024-07-22 at 10 13 46

And release a single stage as owner:

Screenshot 2024-07-22 at 10 17 00

I created a second staged payment and released both stages together:

Screenshot 2024-07-22 at 10 18 58

I was able to release a stage as another user (although only after giving them both `Arbitration` and `Administration`):

Screenshot 2024-07-22 at 10 32 43

And finally was able to release stages via a motion.

Screenshot 2024-07-22 at 10 34 43
Screenshot 2024-07-22 at 10 36 29

File: src/redux/sagas/expenditures/releaseExpenditureStages.ts (hunk 1 above)
Comment (jakubcolony) [reply src/redux/sagas/expenditures/releaseExpenditureStages.ts:83]: You're completely right! Should be fixed now

File: src/redux/sagas/utils/expendituresMulticall.ts [hunk 3]
```diff
@@ -22,6 +22,8 @@ const EXPENDITURESLOTS_SLOT = toB32(BigNumber.from(26));
 const EXPENDITURESLOT_RECIPIENT = toB32(BigNumber.from(0));
 const EXPENDITURESLOT_CLAIMDELAY = toB32(BigNumber.from(1));
 
+const MAX_CLAIM_DELAY_VALUE = constants.MaxUint256;
```
Comment (iamsamgibbs) [inline src/redux/sagas/utils/expendituresMulticall.ts:25]: `constants.MaxUint128` or `constants.MaxUint256`? Previous value matched `constants.MaxUint128` if I remember correctly.

**Review by iamsamgibbs — APPROVED** [review]
Thanks for the fix Jakub, all working as expected now! Able to release as a user with only `arbitration`.

Screenshot 2024-07-25 at 12 21 15

Screenshot 2024-07-25 at 12 23 09

Screenshot 2024-07-25 at 12 21 09

File: src/redux/sagas/utils/expendituresMulticall.ts (hunk 3 above)
Comment (jakubcolony) [reply src/redux/sagas/utils/expendituresMulticall.ts:25]: It should be MaxUint256 according to https://github.com/JoinColony/colonyNetwork/blob/eb3620dbc1232e896f8e4d028889d058363cd0eb/contracts/colony/ColonyDataTypes.sol#L291

File: src/utils/currency/currency.ts [hunk 4]
```diff
@@ -47,7 +47,7 @@ const fetchPriceFromCoinGecko = async ({
 const getCLNYPriceInUSD = async () => {
   // Returns 1 CLNY in terms of USD, 1 CLNY : x USD
   return fetchTokenPriceByAddress({
-    contractAddress: Tokens.Mainnet.Mainnet,
+    contractAddress: Tokens.Mainnet.CLNY,
```
Comment (rdig) [inline src/utils/currency/currency.ts:50]: :+1:

**Review by rdig — APPROVED** [review]
Nicely done. I've tested all 3 ways of releasing a stage: owner, arbitration and motion, all three seem to be working as expected.

Only issue I ran into was not differentiating between staged and staked expenditures extensions, but must be because I am blind :)

Using `owner` permissions:

Screenshot from 2024-07-29 23-32-55
Screenshot from 2024-07-29 23-33-06
Screenshot from 2024-07-29 23-33-28
Screenshot from 2024-07-29 23-35-07
Screenshot from 2024-07-29 23-36-14

Using `arbitration` permissions only:

Screenshot from 2024-07-29 23-37-44
Screenshot from 2024-07-29 23-38-14
Screenshot from 2024-07-29 23-38-27

Using a motion to release: 

Screenshot from 2024-07-29 23-39-23
Screenshot from 2024-07-29 23-39-57
Screenshot from 2024-07-29 23-40-05
Screenshot from 2024-07-29 23-45-14
Screenshot from 2024-07-29 23-45-22
Screenshot from 2024-07-29 23-45-27
Screenshot from 2024-07-29 23-45-39

Nicely done! :tada:

**Pull request merged**